src/clj/puppetlabs/services/master/master_core.clj

```diff
@@ -79,6 +80,17 @@
    jruby-request/wrap-with-error-handling
    ring/wrap-params))
 
+(defn static-file-content-request-handler
+  "Build the main request handler for the /static_file_content endpoint"
+  [get-code-content]
+  (fn [req]
+    (let [environment (get-in req [:params "environment"])
+          code-id (get-in req [:params "code-id"])
+          file-path (get-in req [:params :rest])
+          content-stream (get-code-content environment code-id file-path)]
+      {:status 200
```

Comment:
I wasn't entirely sure what to do in this case. Basically, our `get-code-content` function will rely on the execution of the set script. According to the docstring:

```
Does no exception catching: Being unable to compute code content is considered a fatal error and should be handled by calling code appropriately.
```

Judging from the code, it looks like we're expecting an exception to be thrown by `get-code-content` if the file/environment/code-id doesn't exist (we catch certain exceptions and rethrow as `IllegalStateException`s, and throw an `IllegalStateException` if the script has a non-zero exit code) so that will manifest as an HTTP 500 error and include the Exception's message.

I think we could feasibly change that to return a 404 error, but we'd have to do some exception catching in this method. That wouldn't be too bad, but `get-code-content` will throw an `IllegalArgumentException` in all cases, and there doesn't seem to really be a way to differentiate between when a script failed due to an unexpected error and when a failure occurred because the file/environment/code-id doesn't exist.